### PR TITLE
Laboratory work #4, Bykova Ekaterina - 19FPL2

### PR DESCRIPTION
**Pull request by ffmiil**

### CODE REVIEW

File: lab_4/main.py [hunk 1]
```diff
@@ -4,44 +4,141 @@
 
 from ngrams.ngram_trie import NGramTrie
 
+SENTENCE_END = "<END>"
```
Comment (WhiteJaeger) [inline lab_4/main.py:7]: If it is used only in `tokenize_by_sentence`, the better approach is to place it there.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]
Great work! Just a minor fix and you are done here.

File: lab_4/main.py (hunk 1 above)
Comment (ffmiil) [reply lab_4/main.py:7]: It's used in  NGramTextGenerator either.

File: lab_4/main.py [hunk 2]
```diff
@@ -5,43 +5,141 @@
...
+            raise KeyError
+        return self.storage[word]
 
     def get_word(self, word_id: int) -> str:
-        pass
+        if not isinstance(word_id, int) or isinstance(word_id, bool):
+            raise ValueError
+        if word_id not in self.storage.values():
+            raise KeyError
+        for word, _word_id in self.storage.items():
```
Comment (WhiteJaeger) [inline lab_4/main.py]: If you do not intend to use a var from `.items()`, it is commonly named as `_`.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

**Pull request closed without merging**